### PR TITLE
API_Calls replacement

### PR DESCRIPTION
**Pull request by MitchellBrownNE**
Added the current api_calls.py file to a facade pattern with class SportsAPI and imbedded all the current functions within it.

### CODE REVIEW

File: api_calls.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                print("Error: ", response.status, response.reason)
+                return None
+            
+            data = response.read()
+            json_data = json.loads(data.decode("utf-8"))
+
+            # Loop through the games and find the team id from param team_name
+            for game in json_data['games']:
+                if game['home']['name'] == team_name:
+                    team_id = game['home']['id']
```
Comment (Nobody96103) [inline api_calls.py:70]: This could be replaced with _return game['home']['id'], right?

File: api_calls.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                elif game['away']['name'] == team_name:
+                    team_id = game['away']['id']
+
+        ## Catching exceptions
+        except json.JSONDecodeError as e:
+            print(f"A JSONDecodeError occurred: {str(e)}")
+        except http.client.HTTPException as e:
+            print(f"An exception occurred: {str(e)}")
```
Comment (Nobody96103) [inline api_calls.py:78]: Wouldn't it be better to differentiate between HTTPExceptions and regular exceptions?

**Review by Nobody96103 — COMMENTED** [review]
There isn't really anything to review here - I should have probably just confirmed it was all good before.

File: api_calls.py (hunk 1 above)
Comment (MitchellBrownNE) [reply api_calls.py:70]: This is old code. From Feb 23rd.

File: api_calls.py (hunk 2 above)
Comment (MitchellBrownNE) [reply api_calls.py:78]: This is old code and doesn’t need to be reviewed.

**Pull request merged**